### PR TITLE
Separate API Key auth support generation

### PR DESCRIPTION
**Pull request by benortiz**
## Problem

The generator `stitches:api` forces adding in API Key auth support.

## Solution

Introduce a new generator `stitches:api_migration` that can add in API Key auth support.

### CODE REVIEW

File: lib/stitches/spec/test_headers.rb [hunk 1]
```diff
@@ -34,6 +34,8 @@ def mime_type(options)
   end
 
   def set_authorization_header(headers,options)
+    return nil if Stitches.configuration.disable_api_key_support
```
Comment (benortiz) [inline lib/stitches/spec/test_headers.rb:37]: This is probably my least favorite portion of this. It's not strictly necessary, but having this set the authorization header to nil when api key support is disable seems nice.

Comment (samsm) [reply lib/stitches/spec/test_headers.rb:37]: Let me know if I'm reading this wrong: would this make alternatively make sense?

```diff
}.tap { |headers|
-  set_authorization_header(headers,options)
+  set_authorization_header(headers,options) if Stitches.configuration.disable_api_key_support
}
```

File: lib/stitches/generator_files/spec/features/api_spec.rb.erb [hunk 2]
```diff
@@ -48,6 +48,8 @@ feature "general API stuff" do
     expect(response).to have_api_error(code: "test", message: "OH NOES!")
   end
 
+<%# Remove always false if statement to enable API Key authentication %>
+<% if false %>
```
Comment (samsm) [inline lib/stitches/generator_files/spec/features/api_spec.rb.erb:52]: This kind of jumps out at me but it'll work and I don't have a snappy alternative.

**Review by samsm — APPROVED** [review]
LGTM

File: lib/stitches/generator_files/spec/features/api_spec.rb.erb (hunk 2 above)
Comment (benortiz) [reply lib/stitches/generator_files/spec/features/api_spec.rb.erb:52]: I'm definitely open to changing this now or in the future. Every other "block comment" approach I tried ended up getting mangled with the other ERB.

**Pull request merged**